app/src/main/java/divyansh/tech/animeclassroom/searchAnime/SearchAnimeViewModel.kt

```diff
@@ -22,12 +22,21 @@ class SearchAnimeViewModel @Inject constructor(
     private val _searchAnimeLiveData: MutableLiveData<ResultWrapper<ArrayList<AnimeModel>>> = MutableLiveData()
     val searchAnimeLiveData: LiveData<ResultWrapper<ArrayList<AnimeModel>>> get() = _searchAnimeLiveData
 
+    private val _invalidAnimeSearch: MutableLiveData<Boolean> = MutableLiveData(false)
+    val invalidAnimeSearch: LiveData<Boolean> get() = _invalidAnimeSearch
+
     fun searchAnime(anime: String) = viewModelScope.launch(Dispatchers.IO) {
         when (val response = animeSearchRepo.searchAnimes(anime)) {
             is ResultWrapper.Success ->
             {
                 Log.i("Search", response.data.toString())
-                _searchAnimeLiveData.postValue(ResultWrapper.Success(response.data as ArrayList<AnimeModel>))
+                if(response.data.toString()=="[]"){
```

Comment:
@shubham423 We should not check the list like this. List already contains isEmpty() method, use that. Also, the _invalidAnimeSearch should be true in case of error ie. in the ResultWrapper.Error case.

Reply:
Moreover, we do not need an extra live data just to check if the search contains results or not. See if that can be done inside the existing live data itself.

Reply:
ok @justdvnsh will be correcting the above mentioned points, thanks

Reply:
hi @justdvnsh response.data is of generic data type on which isEmpty() method is not availaible 
by changing the return type of searchAnime() and parsePopularAnimeJson()  funtions from ResultWrapper<*> to ResultWrapper<List<AnimeModel>> i was able to acheive this
is it the right approach ??

Reply:
@shubham423 . Let's look at it. 
The parsePopularAnimeJson() gives us back the ResultWrapper<*> type right ?. That means, it can contain anything inside it. Now the response would either be ResultWrapper.Success() or ResultWrapper.Error() based on the function. Now, inside the viewModel if the list would not be empty.

![image](https://user-images.githubusercontent.com/29144058/135825617-45ac66ee-c461-4e79-a582-687667e14b68.png)


This is the exact error. the response cannot be casted into an ArrayList, right ?. To solve this, we can wrap ``` _searchAnimeLiveData.postValue(ResultWrapper.Success(response.data as ArrayList<AnimeModel>))``` into a try catch block, and then send ResultWrapper.Error() in case of exception.

Reply:
@shubham423 Does that make sense ?

Reply:
@justdvnsh yes it does makes sense

Reply:
@justdvnsh can you please review the changes

Reply:
@shubham423 The code looks fine to me. Just make sure to remove the .idea/ files before commiting.

Reply:
yes removed

Reply:
The Pr still shows me the files @shubham423. You can do one thing, close this PR, and make a new PR with the changes, or if you do not want that, you can follow this link https://stackoverflow.com/a/11124105